Exception when silent hook returns error

**What happened?**  
Having a silent devspace hook which returns an non-null status code results in a panic (see stacktrace at the end).
When changing to `silent: false` devspace aborts properly.

**What did you expect to happen instead?**  
Having a silent devspace hook which returns an non-null status code should either properly return an error or probably silently continue (?).


**How can we reproduce the bug?** (as minimally and precisely as possible)  

My devspace.yaml:
```
[...]
hooks:
  - name: "getting sudo"
    command: false
    events: ["before:deploy"]
    silent: true
...
```

**Local Environment:**  
- DevSpace Version: 6.3.1
- Operating System: linux 
- ARCH of the OS: AMD64
**Kubernetes Cluster:**  
- Cloud Provider: google | aws | azure | other
- Kubernetes Version: [use `kubectl version`]

**Anything else we need to know?**  
stacktrace:
```
$ devspace dev -n test
warn There is a newer version of DevSpace: v6.3.2. Run `devspace upgrade` to upgrade to the newest version.

info Using namespace 'test'
info Using kube context 'minikube'
Execute hook 'getting sudo' at before:deploy
panic: interface conversion: io.WriteCloser is *log.NopCloser, not log.NopCloser

goroutine 77 [running]:
github.com/loft-sh/devspace/pkg/devspace/hook.executeHook({0x27acbf8, 0xc001312bd0}, 0xc0015a40a0, {0x2783268?, 0xc0010fe680?}, 0xc00130eba0, {0x277bd80?, 0xc0013081a0}, {0xc0013201d0, 0xd})
	/Users/runner/work/devspace/devspace/pkg/devspace/hook/hook.go:203 +0x711
github.com/loft-sh/devspace/pkg/devspace/hook.runHook({0x27acbf8, 0xc001312bd0}, 0xc0015a40a0, 0xc00134a120?, {0xc0013201d0, 0xd})
	/Users/runner/work/devspace/devspace/pkg/devspace/hook/hook.go:169 +0x635
github.com/loft-sh/devspace/pkg/devspace/hook.executeSingle({0x27acbf8, 0xc001312bd0}, 0x230b6cc?, {0xc0013201d0, 0xd})
	/Users/runner/work/devspace/devspace/pkg/devspace/hook/hook.go:127 +0x19a
github.com/loft-sh/devspace/pkg/devspace/hook.ExecuteHooks({0x27acbf8, 0xc001312bd0}, 0xc0014aa8e8?, {0xc0014aa9c0?, 0x1, 0x0?})
	/Users/runner/work/devspace/devspace/pkg/devspace/hook/hook.go:90 +0x1c6
github.com/loft-sh/devspace/pkg/devspace/deploy.(*controller).Deploy(0x4306a28, {0x27acbf8, 0xc001312bd0}, {0xc0010fe670, 0x1, 0x0?}, 0xc001350000)
	/Users/runner/work/devspace/devspace/pkg/devspace/deploy/deploy.go:64 +0x11c
github.com/loft-sh/devspace/pkg/devspace/pipeline/engine/pipelinehandler/commands.CreateDeployments({0x27acbf8, 0xc001312bd0}, {0x27a73a0, 0xc00085da40}, {0xc001308150, 0x1, 0x1}, {0x277e3a0?, 0xc000013948})
	/Users/runner/work/devspace/devspace/pkg/devspace/pipeline/engine/pipelinehandler/commands/create_deployments.go:87 +0x68f
github.com/loft-sh/devspace/pkg/devspace/pipeline/engine/pipelinehandler.glob..func8({0x27acbf8, 0xc001312bd0}, {0x27a73a0, 0xc00085da40}, {0xc001308150, 0x1, 0x1})
	/Users/runner/work/devspace/devspace/pkg/devspace/pipeline/engine/pipelinehandler/handler.go:50 +0x18a
github.com/loft-sh/devspace/pkg/devspace/pipeline/engine/pipelinehandler.(*execHandler).handlePipelineCommands.func1()
	/Users/runner/work/devspace/devspace/pkg/devspace/pipeline/engine/pipelinehandler/handler.go:160 +0x43
github.com/loft-sh/devspace/pkg/devspace/pipeline/engine/pipelinehandler.(*execHandler).executePipelineCommand(0x1eee2a0?, {0x2799f88, 0xc00130e810}, {0xc001310168, 0x12}, 0xc0015bbb60?)
	/Users/runner/work/devspace/devspace/pkg/devspace/pipeline/engine/pipelinehandler/handler.go:182 +0x1e2
github.com/loft-sh/devspace/pkg/devspace/pipeline/engine/pipelinehandler.(*execHandler).handlePipelineCommands(0xc00156b130, {0x2799f88, 0xc00130e810}, {0xc001310168, 0x12}, {0xc001308150, 0x1, 0x1})
	/Users/runner/work/devspace/devspace/pkg/devspace/pipeline/engine/pipelinehandler/handler.go:159 +0x6de
github.com/loft-sh/devspace/pkg/devspace/pipeline/engine/pipelinehandler.(*execHandler).ExecHandler(0xc00156b130, {0x2799f88, 0xc00130e810}, {0xc001308140?, 0x2, 0x2})
	/Users/runner/work/devspace/devspace/pkg/devspace/pipeline/engine/pipelinehandler/handler.go:130 +0xad
mvdan.cc/sh/v3/interp.(*Runner).exec(0xc0004f2000, {0x2799ee0?, 0xc001575a80?}, {0xc001308140, 0x2, 0x2})
	/Users/runner/work/devspace/devspace/vendor/mvdan.cc/sh/v3/interp/runner.go:890 +0x63
mvdan.cc/sh/v3/interp.(*Runner).call(0xc0004f2000, {0x2799ee0, 0xc001575a80}, {0x1308150?, 0xc0?}, {0xc001308140?, 0x2, 0x2})
	/Users/runner/work/devspace/devspace/vendor/mvdan.cc/sh/v3/interp/runner.go:886 +0x74b
mvdan.cc/sh/v3/interp.(*Runner).cmd(0xc0004f2000, {0x2799ee0?, 0xc001575a80}, {0x2798520?, 0xc0016140a0?})
	/Users/runner/work/devspace/devspace/vendor/mvdan.cc/sh/v3/interp/runner.go:397 +0x1c5d
mvdan.cc/sh/v3/interp.(*Runner).stmtSync(0xc0004f2000, {0x2799ee0, 0xc001575a80}, 0xc000e110b0)
	/Users/runner/work/devspace/devspace/vendor/mvdan.cc/sh/v3/interp/runner.go:288 +0x2aa
mvdan.cc/sh/v3/interp.(*Runner).stmt(0xc0004f2000, {0x2799ee0?, 0xc001575a80}, 0xc000e110b0)
	/Users/runner/work/devspace/devspace/vendor/mvdan.cc/sh/v3/interp/runner.go:269 +0x176
mvdan.cc/sh/v3/interp.(*Runner).stmts(...)
	/Users/runner/work/devspace/devspace/vendor/mvdan.cc/sh/v3/interp/runner.go:727
mvdan.cc/sh/v3/interp.(*Runner).Run(0xc0004f2000, {0x2799ee0, 0xc001575a80}, {0x2787958?, 0xc001575ac0})
	/Users/runner/work/devspace/devspace/vendor/mvdan.cc/sh/v3/interp/api.go:551 +0x31a
github.com/loft-sh/devspace/pkg/devspace/pipeline/engine.ExecutePipelineShellCommand({0x2799ee0, 0xc001575a80}, {0xc00150cb40?, 0x0?}, {0xc0003b75e0, 0x0, 0x2}, {0xc000054024, 0x8}, 0x0, ...)
	/Users/runner/work/devspace/devspace/pkg/devspace/pipeline/engine/engine.go:88 +0x4cf
github.com/loft-sh/devspace/pkg/devspace/pipeline.(*Job).execute(0xc00156dc80, {0x27acbf8, 0xc00145f320}, {0xc0003b75e0, 0x0, 0x2}, 0x0?, {0x7f3f1ef7a998, 0xc001554a68})
	/Users/runner/work/devspace/devspace/pkg/devspace/pipeline/job.go:121 +0x6b5
github.com/loft-sh/devspace/pkg/devspace/pipeline.(*Job).Run.func1.1()
	/Users/runner/work/devspace/devspace/pkg/devspace/pipeline/job.go:75 +0x3d
github.com/loft-sh/devspace/pkg/util/tomb.(*Tomb).run(0xc00156a960, 0x0?, 0xc001612060)
	/Users/runner/work/devspace/devspace/pkg/util/tomb/tomb.go:189 +0x3b
created by github.com/loft-sh/devspace/pkg/util/tomb.(*Tomb).NotifyGo
	/Users/runner/work/devspace/devspace/pkg/util/tomb/tomb.go:184 +0x131
```

--- a/pkg/devspace/hook/hook.go
+++ b/pkg/devspace/hook/hook.go
@@ -200,7 +200,7 @@ func executeHook(ctx devspacecontext.Context, hookConfig *latest.HookConfig, hoo
 	err := hook.Execute(ctx.WithLogger(hookLog), hookConfig, extraEnv)
 	if err != nil {
 		if hookConfig.Silent {
-			return errors.Wrapf(err, "in hook '%s': %s", ansi.Color(hookName(hookConfig), "white+b"), hookWriter.(logpkg.NopCloser).Writer.(*bytes.Buffer).String())
+			return errors.Wrapf(err, "in hook '%s': %s", ansi.Color(hookName(hookConfig), "white+b"), hookWriter.(*logpkg.NopCloser).Writer.(*bytes.Buffer).String())
 		}
 		return errors.Wrapf(err, "in hook '%s'", ansi.Color(hookName(hookConfig), "white+b"))
 	}

